libs/filamentjs/jsbindings.cpp

```diff
@@ -159,6 +165,36 @@ struct PixelBufferDescriptor {
     std::shared_ptr<driver::PixelBufferDescriptor> pbd;
 };
 
+// Small structure whose sole purpose is to return decoded image data to JavaScript.
+struct DecodedPng {
+    int width;
+    int height;
+    int encoded_ncomp;
+    int decoded_ncomp;
+    BufferDescriptor decoded_data;
+    ~DecodedPng() {
+      if (decoded_data.bd) {
+        stbi_image_free(decoded_data.bd->buffer);
```

Comment:
The lifetime of `decoded_data.bd.buffer` isn't clear to me- it seems like the memory will be freed prematurely, when `DecodedPng` goes out of scope, even though the buffer gets passed to `Filament.PixelBuffer`:

```
  const pixelbuffer = Filament.PixelBuffer(decodedpng.data.getBytes(), pbformat, pbtype);
```

Reply:
Excellent catch, fixed in a subsequent CL.  I had done this while diagnosing a memory leak.

Incidentally I need to write some leak tests, the memory management is very tricky since embind isn't designed to work with raw pointers.